### PR TITLE
Calculator Component

### PR DESCRIPTION
**Pull request by JAAR91**
- Create Calculator component UI
- Render it using App.js
- Used class-based components

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,29 +1,10 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+  ReactDOM.render(<Calculator />,
+    document.getElementById('container'));
```
Comment (DeliceLydia) [inline src/App.js]: - [x] App function should be class based please work on that

File: src/index.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
 import './index.css';
-import App from './App';
 import reportWebVitals from './reportWebVitals';
```
Comment (DeliceLydia) [inline src/index.js]: - [x] This are boilerplate you don't  need them in your application
please delete them

File: src/index.js [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
+
+// ReactDOM.render(
+//   <React.StrictMode>
+//     <App />
+//   </React.StrictMode>,
+//   document.getElementById('root'),
+// );
 
 // If you want to start measuring performance in your app, pass a function
 // to log results (for example: reportWebVitals(console.log))
```
Comment (DeliceLydia) [inline src/index.js]: - [x] Delete even all the commented codes to make your codes
cleaner

File: public/index.html [hunk 4]
```diff
@@ -24,7 +25,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+    <title>Math Magicians</title>
   </head>
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
```
Comment (DeliceLydia) [inline public/index.html:31]: - [x] In the public folder you can keep only `index.html` others are boilerplate
they are not needed now

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @JAAR91,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._


### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (JAAR91) [reply src/App.js]: This code is from the develop branch which is not related to the branch I am submitting on this pull request.

File: src/App.js [hunk 5]
```diff
@@ -1,29 +1,10 @@
-import logo from './logo.svg';
+import ReactDOM from 'react-dom';
 import './App.css';
+import Calculator from './components/Calculator';
 
 function App() {
```
Comment (Dipeshtwis) [inline src/App.js:52]: - [x] What the previous reviewer said is not about the deleted code, it's about the component itself. In this milestone, The `App.js` component should be class-based. The objective is to practice the class-based React component.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @JAAR91, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**

-------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by JAAR91** [comment]
For the next Code Reviewer:

The class components feature was used inside the Calculator.js. That is why App.js was not modified.

Thanks.
Have a nice day.

File: src/App.js (hunk 5 above)
Comment (Davidosky007) [reply src/App.js:52]: * The requested changes by the previous code reviewer is required for this project to be approved, it is stated in the project requirement to use class-based components for this project kindly fix this 🙏 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @JAAR91 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by mimipeshy** [comment]
### Hello @JAAR91  👋,

_This is an additional review as a result of the second opinion request._

As the last CR pointed out, the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md) you are required to use class components whether the file was modified or not 😄in the bigger picture, all files will be class-based and we want to ensure that all files are converted, so we do that one step at a time. Kindly work on the changes required and submit them for another review. Cheers 😄 
Keep rocking 🚀!

**Pull request closed without merging**